controller/api/destination/endpoint_translator.go

```diff
@@ -565,10 +567,7 @@ func createWeightedAddrForExternalWorkload(
 	opaquePort = opaquePort || address.OpaqueProtocol
 
 	if forceOpaqueTransport || opaquePort {
```

Comment:
Should we do a similar treatment for pods as well? I am not too sure. Pods should have this port available in the spec if they are injected.

Reply:
for pods we read it from the proxy's env (i.e. as it is in injected)